### PR TITLE
feat(be/search engine): create clients for dblp, scholar

### PR DESCRIPTION
**Pull request by jankrokos**

### CODE REVIEW

File: rms/search_engine/clients.py [hunk 1]
```diff
@@ -71,3 +74,59 @@ def _build_abstract_keywords_criteria(abstract_keywords: list[str] | None) -> st
...
+
+class ScholarClient(ABC):
+
+    def __init__(self):
+        self.sections = ["basics", "indices"]
+
+
+class ScholarAuthorSearchApi(ScholarClient):
+
+    def __init__(self):
```
Comment (jsyska) [inline rms/search_engine/clients.py:121]: Not needed when initialization behavior is not changed. Init method from super class will be called automatically

File: rms/search_engine/clients.py [hunk 2]
```diff
@@ -71,3 +74,59 @@ def _build_abstract_keywords_criteria(abstract_keywords: list[str] | None) -> st
...
+        self.headers = {
+            "httpAccept": "application/json",
+        }
+        self.dblp_search_api_endpoint = self.settings.dblp_search_endpoint
+
+
+class DblpAuthorSearchApi(DblpClient):
+
+    def __init__(self):
+        super().__init__()
```
Comment (jsyska) [inline rms/search_engine/clients.py:92]: Not needed when initialization behavior is not changed.

File: rms/search_engine/views.py [hunk 3]
```diff
@@ -1,10 +1,33 @@
...
+
+    return dblp_author
+
+
+@router.get("/get_author_scholar")
+async def get_author_scholar(body: ScholarAuthorSearchBody) -> ScholarAuthorResponse:
+    scholar_author = await get_author_scholar_service(body)
+
+    if not scholar_author:
+        raise NotFound
```
Comment (jsyska) [inline rms/search_engine/views.py:31]: i would return response with code 200 and empty string - not code 404

File: rms/search_engine/views.py [hunk 4]
```diff
@@ -1,10 +1,33 @@
...
 @router.post("/discover_reviewers")
 async def search_scopus(body: SearchBody) -> SearchResponse:
     return await search_scopus_service(body)
+
+
+@router.get("/get_author_dblp")
+async def get_author_dblp(body: DblpAuthorSearchBody) -> DblpAuthorResponse:
+    dblp_author = await get_author_dblp_service(body)
+
+    if not dblp_author:
```
Comment (jsyska) [inline rms/search_engine/views.py:20]: i would return response with code 200 and empty string - not code 404

**Review by jsyska — CHANGES_REQUESTED** [review]

**Review by CreativePhilip — CHANGES_REQUESTED** [review]

File: rms/search_engine/clients.py (hunk 1 above)
Comment (jankrokos) [reply rms/search_engine/clients.py:121]: fixed

File: rms/search_engine/clients.py (hunk 2 above)
Comment (jankrokos) [reply rms/search_engine/clients.py:92]: fixed

File: rms/search_engine/views.py (hunk 4 above)
Comment (jankrokos) [reply rms/search_engine/views.py:20]: fixed

File: rms/search_engine/views.py (hunk 3 above)
Comment (jankrokos) [reply rms/search_engine/views.py:31]: fixed

**Review by CreativePhilip — APPROVED** [review]

**Review by jsyska — APPROVED** [review]

**Pull request merged**